src/registered_functions.jl

```diff
@@ -132,18 +132,18 @@ function expand_registered_functions(expr)
                ((args[1])^args[5] + (args[2])^args[5] + (args[4])^args[5])
     end
     for i in 1:length(args)
-        args[i] = expand_registered_functions(args[i])
+        args[i] = expand_registered_functions!(args[i])
     end
     return expr
 end
 # If applied to a Reaction, return a reaction with its rate modified.
 function expand_registered_functions(rx::Reaction)
-    Reaction(expand_registered_functions(rx.rate), rx.substrates, rx.products,
+    Reaction(expand_registered_functions!(deepcopy(rx.rate)), rx.substrates, rx.products,
```

Comment:
Is this appropriate use of `deepcopy`? I could not come up with a better way to do it. If we really want to avoid it it would be possible to first check the expression whether there are any registered functions, and only apply the function in this case

Reply:
Also, I think technically this by could be rewritten using `@set` (since expanding functions really shouldn't change anything more of the reaction), however, I was reluctant to introduce any more changes than necessary.

Reply:
It would be more memory efficient to first check if the given expression would be expanded, and only then deepcopy and expand, but otherwise this is fine as a less memory efficient version.

Reply:
(Note, I'm not saying this needs to be done in this PR, just that it would be good to switch to that eventually.)